test/e2e/ingress.go

```diff
@@ -274,13 +278,19 @@ func (cont *IngressController) create() {
 		Expect(err).NotTo(HaveOccurred())
 	}
 	rc := rcFromManifest(cont.rcPath)
-	existingRc, err := cont.c.ReplicationControllers(api.NamespaceSystem).Get(lbContainerName)
+
+	listOpts := api.ListOptions{LabelSelector: labels.SelectorFromSet(labels.Set(clusterAddonLBLabels))}
+	existingRCs, err := cont.c.ReplicationControllers(api.NamespaceSystem).List(listOpts)
 	Expect(err).NotTo(HaveOccurred())
+	if len(existingRCs.Items) == 0 {
```

Comment:
Since you take the 0th item in the list, this assertion should be `if len(existingRCs.Items) != 1` to detect cases where more than one RC matches the label query.